### DISCUSSION TITLE
table sorting by multiple columns

### DISCUSSION THREAD

**Discussion by dreamgear** [element-plus/element-plus#14160] [Q&A]
The docs say:

> Set table attribute default-sort to determine default sort column and order. To apply your own sorting rules, use sort-method or sort-by

What is the syntax for specifying multiple sort fields, and where is it documented ?  Or where are sort-method and sort-by documented ?

Thanks in advance.

**Comment by vela666** [comment]
![image](https://github.com/element-plus/element-plus/assets/69613314/b0cd09c7-a673-408f-b6c0-9547c8daa6cb)

This is my implementation method, you can refer to it
[demo](https://element-plus.run/#eyJzcmMvQXBwLnZ1ZSI6IiA8dGVtcGxhdGU+XG4gIDxlbC10YWJsZVxuICAgIHJlZj1cInRhYmxlTGlzdFwiXG4gICAgOmRhdGE9XCJ0YWJsZURhdGFcIlxuICAgIGJvcmRlclxuICAgIDpoZWFkZXItY2VsbC1jbGFzcy1uYW1lPVwiaGFuZGxlSGVhZGVyQ2xhc3NcIlxuICAgIEBzb3J0LWNoYW5nZT1cImhhbmRsZVNvcnRDaGFuZ2VcIlxuICA+XG4gICAgPGVsLXRhYmxlLWNvbHVtbiBzb3J0YWJsZSBwcm9wPVwiZGF0ZVwiIGxhYmVsPVwiRGF0ZVwiIHdpZHRoPVwiMTgwXCIgLz5cbiAgICA8ZWwtdGFibGUtY29sdW1uIHNvcnRhYmxlIHByb3A9XCJuYW1lXCIgbGFiZWw9XCJOYW1lXCIgd2lkdGg9XCIxODBcIiAvPlxuICAgIDxlbC10YWJsZS1jb2x1bW4gc29ydGFibGUgcHJvcD1cImFkZHJlc3NcIiBsYWJlbD1cIkFkZHJlc3NcIiAvPlxuICA8L2VsLXRhYmxlPlxuPC90ZW1wbGF0ZT5cblxuPHNjcmlwdCBzZXR1cD5cbmltcG9ydCB7IHJlYWN0aXZlIH0gZnJvbSAndnVlJ1xuXG5jb25zdCB0YWJsZURhdGEgPSBbXG4gIHtcbiAgICBkYXRlOiAnMjAxNi0wNS0wMycsXG4gICAgbmFtZTogJ1RvbScsXG4gICAgYWRkcmVzczogJ05vLiAxODksIEdyb3ZlIFN0LCBMb3MgQW5nZWxlcycsXG4gIH0sXG4gIHtcbiAgICBkYXRlOiAnMjAxNi0wNS0wMicsXG4gICAgbmFtZTogJ1RvbScsXG4gICAgYWRkcmVzczogJ05vLiAxODksIEdyb3ZlIFN0LCBMb3MgQW5nZWxlcycsXG4gIH0sXG4gIHtcbiAgICBkYXRlOiAnMjAxNi0wNS0wNCcsXG4gICAgbmFtZTogJ1RvbScsXG4gICAgYWRkcmVzczogJ05vLiAxODksIEdyb3ZlIFN0LCBMb3MgQW5nZWxlcycsXG4gIH0sXG4gIHtcbiAgICBkYXRlOiAnMjAxNi0wNS0wMScsXG4gICAgbmFtZTogJ1RvbScsXG4gICAgYWRkcmVzczogJ05vLiAxODksIEdyb3ZlIFN0LCBMb3MgQW5nZWxlcycsXG4gIH0sXG5dXG5jb25zdCBpbml0VmFsID0gKCkgPT4ge1xuICByZXR1cm4ge1xuICAgIC8vIOaYr+WQpua3u+WKoOaOkuW6j+agt+W8jyhXaGV0aGVyIHRvIGFkZCBhIHNvcnQgc3R5bGUpXG4gICAgbWFyazogZmFsc2UsXG4gICAgb3JkZXI6IFtdLFxuICB9XG59XG5cbmNvbnN0IHN0YXRlID0gcmVhY3RpdmUoaW5pdFZhbCgpKVxuXG5jb25zdCBjbGVhck9yZGVyID0gKCkgPT4ge1xuICBPYmplY3QuYXNzaWduKHN0YXRlLCBpbml0VmFsKCkpXG59XG5cbmNvbnN0IGdldFRhYmxlRGF0YSA9ICgpID0+IHt9XG5cbmNvbnN0IGhhbmRsZU9yZGVyQ2hhbmdlID0gKHsgbXVsdGlPcmRlciwgcHJvcGVydHksIGluZGV4IH0pID0+IHtcbiAgY29uc3QgdG1wID0gbXVsdGlPcmRlciA9PT0gJ2Rlc2NlbmRpbmcnID8gJ0RFU0MnIDogJ0FTQydcbiAgc3RhdGUub3JkZXIgPSBzdGF0ZS5vcmRlci5maWx0ZXIoKGl0ZW0pID0+ICFpdGVtLmluY2x1ZGVzKHByb3BlcnR5KSlcbiAgLy8g5Yig6Zmk5Li6bnVsbOeahOaOkuW6j1xuICAvLyBwcm9wOltcInJldmVudWUgQVNDXCIsXCJkYXkgREVTQ1wiXSBhc2NlbmRpbmcgZGVzY2VuZGluZ1xuICBzdGF0ZS5vcmRlci5wdXNoKGAke3Byb3BlcnR5fSAke3RtcH1gKVxuICAvLyDkuI3mjpLluo/liJnliKDpmaQgKG5vdCBzb3J0ZWQpXG4gIGlmICghbXVsdGlPcmRlcikge1xuICAgIHN0YXRlLm9yZGVyLnBvcCgpXG4gIH1cblxuICBjb25zb2xlLmxvZyhzdGF0ZS5vcmRlciwgcHJvcGVydHkpXG4gIHN0YXRlLm1hcmsgPSB0cnVlXG4gIGdldFRhYmxlRGF0YSgpXG59XG5cbi8vIOeCueWHu+aOkuW6j+aXtuinpuWPkVxuY29uc3QgaGFuZGxlU29ydENoYW5nZSA9ICh7IGNvbHVtbiwgcHJvcCwgb3JkZXIgfSkgPT4ge1xuICBpZiAoIWNvbHVtbi5tdWx0aU9yZGVyKSB7XG4gICAgY29sdW1uLm11bHRpT3JkZXIgPSAnYXNjZW5kaW5nJ1xuICB9IGVsc2UgaWYgKGNvbHVtbi5tdWx0aU9yZGVyID09PSAnYXNjZW5kaW5nJykge1xuICAgIGNvbHVtbi5tdWx0aU9yZGVyID0gJ2Rlc2NlbmRpbmcnXG4gIH0gZWxzZSB7XG4gICAgY29sdW1uLm11bHRpT3JkZXIgPSBudWxsXG4gIH1cbiAgaGFuZGxlT3JkZXJDaGFuZ2UoY29sdW1uKVxufVxuY29uc3QgaGFuZGxlSGVhZGVyQ2xhc3MgPSAoeyBjb2x1bW4gfSkgPT4ge1xuICBjb25zb2xlLmxvZyhjb2x1bW4ubXVsdGlPcmRlciwgY29sdW1uLnByb3BlcnR5LCBzdGF0ZS5tYXJrKVxuICBjb2x1bW4ub3JkZXIgPSBzdGF0ZS5tYXJrID8gY29sdW1uLm11bHRpT3JkZXIgOiBudWxsXG59XG5cbmRlZmluZU9wdGlvbnMoe1xuICBuYW1lOiAnRGVtbycsXG59KVxuPC9zY3JpcHQ+IiwiaW1wb3J0LW1hcC5qc29uIjoie1xuICBcImltcG9ydHNcIjoge31cbn0iLCJ0c2NvbmZpZy5qc29uIjoie1xuICBcImNvbXBpbGVyT3B0aW9uc1wiOiB7XG4gICAgXCJhbGxvd0pzXCI6IHRydWUsXG4gICAgXCJjaGVja0pzXCI6IHRydWUsXG4gICAgXCJqc3hcIjogXCJwcmVzZXJ2ZVwiLFxuICAgIFwidGFyZ2V0XCI6IFwiRVNOZXh0XCIsXG4gICAgXCJtb2R1bGVcIjogXCJFU05leHRcIixcbiAgICBcIm1vZHVsZVJlc29sdXRpb25cIjogXCJCdW5kbGVyXCIsXG4gICAgXCJhbGxvd0ltcG9ydGluZ1RzRXh0ZW5zaW9uc1wiOiB0cnVlLFxuICAgIFwidHlwZXNcIjogW1wiZWxlbWVudC1wbHVzL2dsb2JhbC5kLnRzXCJdXG4gIH0sXG4gIFwidnVlQ29tcGlsZXJPcHRpb25zXCI6IHtcbiAgICBcInRhcmdldFwiOiAzLjNcbiAgfVxufVxuIiwiX28iOnt9fQ== "demo")

**Reply by tom-field** [reply]
nice idea

**Reply by ta-soft** [reply]
any chance of suggesting a version that would work for <script lang="ts" setup> pages?
